lib/id3c/cli/command/etl/presence_absence.py

```diff
@@ -80,16 +80,26 @@ def etl_presence_absence(*, action: str):
                 LOG.info(f"Processing presence_absence group {group.id}")
 
                 # Samplify will now send documents with a top level key
-                # "samples". So we have three different top level keys:
-                # "store", "Update" and "samples"
-                #   -Jover, 07 Nov 2019
+                # "samples". The new format also includes a "chip" key for each
+                # sample which is then included in the unique identifier for
+                # each presence/absence result
+                #   -Jover, 14 Nov 2019
                 try:
                     received_samples = group.document["samples"]
-                except KeyError:
-                    try:
-                        received_samples = group.document["store"]["items"]
-                    except KeyError:
-                        received_samples = group.document["Update"]
+                except KeyError as error:
+                    # Skip documents in the old format because they do not
+                    # include the "chip" key which is needed for the
+                    # unique identifier for each result.
+                    #   -Jover, 14 Nov 2019
+                    if (group.document.get("store") is not None or
+                        group.document.get("Update") is not None):
+
+                        LOG.info("Skipping presence_absence record that is in old format")
+                        mark_processed(db, group.id)
+                        continue
+
+                    else:
+                        raise error from None
```

Comment:
My first thought was that it'd be slightly more robust/future-proof to skip each _sample_ `if not received_sample.get("chip")` instead of skipping each _document_ based on the container/collection format (which assumes the container format is fundamentally linked to the inclusion of `chip`).

Upon further consideration though, I think I prefer the way you have it :smile:, because it will fail loudly if we get a sample that doesn't include a `chip`.

Reply:
I hadn't thought that far ahead, but glad my way works 😆